src/credentials_rotators/npm/stacks/common_stack.py

```diff
@@ -0,0 +1,135 @@
+        used for authentication.
+        Args:
+            rotator_lambda (Function): The lambda function used for rotating a secret
+            secret_configs (Dictionary): list of configurations for static secrets used for authentication by rotator_lambda
+        """
+        for secret_config in secret_configs:
+            secret_arn = get_secret_arn(secret_config)
+            rotator_lambda.add_to_role_policy(PolicyStatement(effect=Effect.ALLOW,
+                                                              resources=[secret_arn],
+                                                              actions=["secretsmanager:GetSecretValue"]
+                                                              )
+                                              )
+
+    def configure_secret_rotation(self, rotator_lambda, secret_config, duration):
+        """
+        Adds the rotator_lambda to the secret referenced using secret_id in the secrets_config.json.
+        Args:
+            rotator_lambda (Function): The lambda function used for rotating a secret
+            secret_config (Dictionary): The configuration for the secret specified in secrets_config.json
+            duration (Duration): The rotation time interval
+        """
+        secret_arn = get_secret_arn(secret_config)
+        secret_key = get_secret_key(secret_config)
+        secret = Secret.from_secret_complete_arn(self,
+                                                 secret_key,
+                                                 secret_complete_arn=secret_arn)
+        secret.add_rotation_schedule(id=f'{secret_key}-rotator',
+                                     automatically_after=duration,
+                                     rotation_lambda=rotator_lambda)
+
+    def enable_cloudwatch_alarm_notifications(self, rotator_lambda, secret_id):
+        """
+        Adds a cloudwatch alarm to monitor the error metrics.
+        Subscribes the given emails to receive email alerts.
+        Args:
+            rotator_lambda (Function): The lambda function used for rotating a secret
+            secret_id (string): The identifier corresponding to the secret in the secrets_config.json file
+        """
+        # create an sns topic for the rotator lambda monitoring
+        alarm_sns_topic_id = f'{secret_id}_alarm_sns_topic'
+        alarm_sns_topic = Topic(self,
+                                alarm_sns_topic_id)
+
+        # subscribe the given emails to the sns topic
+        subscription_emails = get_alarm_subscriptions(secret_id)
+        for email in subscription_emails:
+            alarm_sns_topic.add_subscription(EmailSubscription(email))
+
+        # add errors metric alarm to rotator lambda
+        # should send and email notification if the errors metric >= threshold every single time(evaluation_periods)
+        errors_alarm_id = f'{secret_id}_errors_alarm'
+        errors_alarm = rotator_lambda.metric_errors().create_alarm(self,
```

Comment:
so this alarms after one error is raised just to be clear?

Reply:
Yes. The threshold is set to 1 and it should trigger the alarm when no.of.errors >= 1 (GreaterThanOrEqualTo is the default comparison operator)